### DISCUSSION TITLE
Next js component not updated with new data when I change route between pages?

### DISCUSSION THREAD

**Discussion by nekmart** [vercel/next.js#47142] [Help]
### Summary

Hello, I create a nextjs project and I deploy it to vercel. but I am facing one problem, I create  new data or I am deleting some data, after change route and come back previous route, I can see that previous data not updated. Please see the video-

[6091291484416378925.webm](https://user-images.githubusercontent.com/111189053/225228493-39b36d45-aa7e-4524-b692-e12fbbf26710.webm)

In this video, we can see new category not coming in the list. If I hard refresh my page or open it into incognito then then I can see the updated data.

Here is live link-

https://dashboard-ten-sand.vercel.app/

email or phone- 01611994403
password- siam1980

Here is my project-  https://github.com/nekmart/admin-dashboard

Please anybody gives me some hints why and where this problem occured?

**Comment by icyJoseph** [comment]
Hi,

You are using a PWA.

The most popular `next-pwa` uses https://developer.chrome.com/docs/workbox/modules/workbox-strategies/#stale-while-revalidate, stale while revalidate, by default.

This means that the worker intercepts requests and serves them from its cache. This is why you have to refresh or do other tricks to get the data to show. In other words, stale-while-revalidate, defeats the purpose of GetServerSideProps.

You need to change to, https://developer.chrome.com/docs/workbox/modules/workbox-strategies/#network-first-network-falling-back-to-cache, so that the worker always lets the request through, updating the cache as it goes, and only if the network fails, data is served from the cache.

I'll come back with some code in a moment...

You can refer to this older answer: https://github.com/vercel/next.js/discussions/42405#discussioncomment-4046461

Reposting code, because why not:

```js
/** @type {import('next').NextConfig} */
const runtimeCaching = require("next-pwa/cache");

const nextDataIndex = runtimeCaching.findIndex(
  (entry) => entry.options.cacheName === "next-data"
);

if (nextDataIndex !== -1) {
  runtimeCaching[nextDataIndex].handler = "NetworkFirst";
} else {
  throw new Error("Failed to find next-data object in runtime caching");
}

const withPWA = require("next-pwa")({
  dest: "public",
  runtimeCaching,
  register: true,
  skipWaiting: true,
  buildExcludes: [/middleware-manifest.json$/],
  disable: process.env.NODE_ENV === 'development'
})

module.exports = withPWA({
  reactStrictMode: false
});
```
